src/test/java/org/broadinstitute/hellbender/utils/clipping/ReadClipperUnitTest.java

```diff
@@ -444,4 +442,22 @@ public void testSoftClipByReferenceCoordinates() {
             }
         }
     }
+
+    // Test fix for https://github.com/broadinstitute/gatk/issues/3466
+    @Test
+    public void testHardClipSoftClippedBasesResultsInEmptyReadDontSetNegativeStartPosition() {
+        final GATKRead originalRead = ArtificialReadUtils.createArtificialRead(TextCigarCodec.decode("170H70S"));
+        // It's important that the read be near the start of the contig for this test, to test
```

Comment:
Looking at the bug, this is technically not correct. The code was always producing bogus read start positions for empty reads before, its just that they never got error checked unless they fell off the contig. Either way this looks like a good test.

Reply:
What the comment means is that the test case only fails before the patch if the read is near the start of the contig. Otherwise you don't end up with a negative start, and the corrupt read gets replaced by a new "empty" read.